docs/troubleshooting.md

```diff
@@ -1,79 +1,165 @@
+If you are new to the project, you may find it helpful to read the [introduction](./what-is-hls.md) page, which explains some of the terminology used on this page.
 
-### Preprocessor
-HLS is not yet able to find project preprocessors, which may result in `could not execute: <preprocessor>` errors. This problem is
-tracked in https://github.com/haskell/haskell-language-server/issues/176 and originally comes from https://github.com/mpickering/hie-bios/issues/125
+## Getting help
 
-As a workaround, you need to ensure the preprocessor is available in the path (install globally with Stack or Cabal, provide in `shell.nix`, etc.).
+### Where to ask
 
-Example with `tasty-discover`:
+Many of the developers are active on [IRC](https://web.libera.chat/?channels=#haskell-language-server).
+However, the most direct way to get help is to open an [issue](https://github.com/haskell/haskell-language-server/issues).
 
-```haskell
-{-# OPTIONS_GHC -F -pgmF tasty-discover #-}
-```
+If you can diagnose whether a problem is with HLS or with the client that you are using, then it is helpful to open an issue in the appropriate repository.
+But this can be tricky, and if you're not sure then you can always open one in the HLS repository and we'll help you figure out what's going on.
 
-This returns an error in HLS if 'tasty-discover' is not in the path: `could not execute: tasty-discover`.
+### What to include
 
-### Problems with multi component (tests suites, executables, benchmarks) support using stack
+Please try and give us as much information as you can!
+In particular, the more you can diagnose the problem, the better.
 
-Due to some limitations in the interaction between HLS and stack, there are some issues in projects with a main library and executables, test suites or benchmarks:
-- The stack project has to be built succesfully *before* loading it with HLS to get components other than the library work.
-- Changes in the library are not automatically propagated to other components, especially in the presence of errors in the library, so you have to restart HLS to get those components correctly loaded. The usual symptom is the editor showing errors like `Could not load module ...` or `Cannot satisfy -package ...`
-- See https://github.com/haskell/haskell-language-server/issues/366 for more info about.
-## Common issues
+## Basic diagnosis steps
 
-### Problems with dynamic linking
+### Quick diagnosis checklist
 
-As haskell-language-server prebuilt binaries are statically linked, they don't play well with projects using dynamic linking.
-An usual symptom is the presence of errors containing `unknown symbol` and it is typical in arch linux, where a dynamically linked version of ghc is used.
+1. Is the server binary running at all? (check process monitors etc.) If not, there may be a client or installation problem.
+2. Is the client connecting to the server? (check for indicators in the editor etc.) If not, there may be a client problem.
```

Comment:
Should we explain this further since at least in neovim there is no indicators by default but there is a command that user has to execute manually?

Reply:
I guess each of these probably needs a paragraph...

Reply:
I might leave the first version light on detail, but I would very much welcome contributions from people who know e.g. the vim setup better. At least it would be nice to have a link to the vim client documentation in here.

Reply:
Ok. I guess I'll just wait until this PR is merged with whatever content you see fit and will create another one with vim details if necessary.